lib/iris/coords.py

```diff
@@ -777,6 +775,24 @@ def cube_dims(self, cube):
         """
         return cube.ancillary_variable_dims(self)
 
+    def xml_element(self, doc, split=None):
```

Comment:
is there a reason this method returns a different signature (element, attributes) vs. (element,) depending on the param value `split`?  different use cases?

Reply:
The default signature `xml_element(doc) => (element,)` is different to the default signature of e.g. `Coord#xml_element => (element, attributes)`.  The api seems inconsistent throughout this file, is this expected?

Reply:
@jamesp Yes, this is intentional, and I couldn't see a neater way around it.

The issue is being forced by the inheritance hierarchy and the fact that the `xml_element` method in the **final** inheritance call chain **must** return an `xml.dom.minidom.Element`.

Just be to explicit we have the following:
```
CellMethod
`-- xml_element()

_DimensionalMetadata
|-- AncillaryVariable
|   |-- CellMeasure
|   |   `-- xml_element()
|   `-- xml_element()
|-- Coord
|   |-- AuxCoord
|   |   `-- xml_element()
|   |-- DimCoord
|   |   `-- xml_element()
|   `-- xml_element()
`-- xml_element()
```

So `AncillaryVariable.xml_element` by default requires to return an `xml.dom.minidom.Element` if it is **directly called**, otherwise, if `AncillaryVariable.xml_element` is called **indirecty** as part of the `super` in `CellMeasure.xml_element`, `CellMeasure` requires `AncillaryVariable.xml_element` to return `(xml.dom.minidom.Element, dict)` instead as it still has has to mix-in the `measure` attribute before finally calling `xml.dom.minidom.Element.setAttribute`.

Why are we doing this now?

Well, basically, for Python `<3.8` the `__str__`/`__repr__` of a `xml.dom.minidom.Element` would show its elements "for free" in alphabetical order... for Python `>=3.8` it now shows its elements in assigned order - which breaks all our `cml` tests i.e., I'm guessing this is a direct side-effect of `dict`s behaving more like `OrderedDict`s.

So the deal is that we need **all** the attributes (`key`-`values` pairs) that we're ever going to set on the `xml.dom.minidom.Element` **all** at once before doing so, but sort them by `key` first, to get the same behaviour pre-3.8.

The fact that `CellMeasure` is a sub-class of `AncillaryVariable` is forcing this dual return type behaviour from `AncillaryVariable`.

Does this help?

Reply:
BTW I'm not wedded to the `split` kwarg as a name. If you agree on my strategy of doing all this, and the purpose of the `split` kwarg... then please suggest a rename for it, if you can think of something better.

Reply:
I guess I could be a good citizen here and add all the missing doc-strings for the `xml_element` methods documenting the args, kwargs and return types...

Reply:
In fact, I'm just going to go ahead an just do that... why not :wink:

Reply:
Done